src/GitVersionCore/Extensions/GitExtensions.cs

```diff
@@ -31,7 +31,7 @@ public static string NameWithoutOrigin(this Branch branch)
         /// <summary>
         /// Checks if the two branch objects refer to the same branch (have the same friendly name).
         /// </summary>
-        public static bool IsSameBranch(this Branch branch, Branch otherBranch)
+        public static bool IsSameBranch(this IBranch branch, IBranch otherBranch)
```

Comment:
Can `IsSameBranch` be removed in favor of `IBranch.Equals` or simply `==`?

Reply:
Sure, makes sense, I will change that in a subsequent PR